blob/oc_test.go

```diff
@@ -55,11 +55,11 @@ func TestOpenCensus(t *testing.T) {
 	const provider = "gocloud.dev/blob/memblob"
 
 	diff := octest.Diff(te.Spans(), te.Counts(), "gocloud.dev/blob", provider, []octest.Call{
-		{"NewWriter", gcerrors.OK},
-		{"NewRangeReader", gcerrors.OK},
-		{"Attributes", gcerrors.OK},
-		{"Delete", gcerrors.OK},
-		{"NewRangeReader", gcerrors.NotFound},
+		{Method: "NewWriter", Code: gcerrors.OK},
```

Comment:
We control `octest`, so it shouldn't be necessary to provide struct literal keys. This is a case of vet being too picky.

Reply:
Is there a bug open for `go vet` about this? If they are going to fix it, fine, but if not I'd rather err on the side of making `go vet` happy unless there's a real reason not to.

Reply:
Especially since these existing warnings mask real ones; for example, I didn't notice the `Example`-related govet warnings in my editor because of all of the `octest` ones.

Reply:
The reason not to is that it leads to less readable code. I agree the example errors are useful, but most of these make the code worse.
Let's do what Jean did for the cloud.google.com/go repo, and grep out the messages we don't want. See https://code.googlesource.com/gocloud/+/refs/heads/master/internal/kokoro/vet.sh. 

Or we could run `go vet -composites=false`, since all the false positives are due to that one check.

Reply:
Honestly, I don't find the code less readable.

I'm not planning on adding a Travis check based on `go vet`, although I guess we could. This is mostly so that I see warnings when I run it locally (my editor does it for me). I'd really prefer not to have to figure out how to configure it to run something other than the default.

If you think these warnings are invalid, that's a discussion for `go vet` owners; I poked around a bit and found https://github.com/golang/go/issues/6820 which seems to imply it's unlikely to be changed.

For example, what happens if/when we start supporting drivers in other repos; we'll likely have to make `internal/octest` public so that others can use it. Then changes to the struct could have compatibility problems.

Reply:
@zombiezen 	also had some reasons why the "composites" check is legit.

Reply:
@jba is aware of all of them AFAIK. Protobufs potentially changing internal layout was one that plagued the cloud.google.com/go repo. Crossing packages, even if we're in control of both packages, can be a bit dicey. If it really affects readability, then we should add a constructor function that builds the shorter parameter list.